packages/craco/scripts/build.js

```diff
@@ -6,10 +7,10 @@ const { loadCracoConfig } = require("../lib/config");
 const { overrideWebpack } = require("../lib/features/webpack");
 
 log("Override started with arguments: ", process.argv);
-log("For environment: ", process.env.NODE_ENV);
+log("For environment: ", buildEnv);
 
 const context = {
-    env: process.env.NODE_ENV,
+    env: buildEnv,
```

Comment:
I am not sure the environnement should be different between the process and the craco context.

What is your use case?

Reply:
We use styled-component and we're making E2E tests with Cypress. 
To do this we need to add an attribute on our elements : `data-test-id`
So, what we're doing on our CI : 
- building app
- serve static files with `serve`
- launch our Cypress tests

What i want : when i set the NODE_ENV to staging is keep these `data-test-id` to run my tests and when i set it to production i want to remove them.

Reply:
Got it.

When you run your test using a staging environment, do you need to compile your react app with all the production optimizations?

If not, updating for the following could solve your use case:

process.env.NODE_ENV = process.env.NODE_ENV || "production"

Reply:
You got it 👍 
It was the solution i tried first but there is a problem. Webpack need the NODE_ENV to be set to "production"

Reply:
Hi @jean-smaug 

I dont think you should mess with the NODE_ENV, as you said, you need it to be set to "production".

Adding an additional env var is out of scope of craco. It can be done outside of craco.

Reply:
ok np, we can close this one